### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Layrovell**
[DEMO LINK](https://Layrovell.github.io/react_add-todo-form/)

### CODE REVIEW

**Review by Serhii-Alieksieievych — APPROVED** [review]

File: src/App.js [hunk 1]
```diff
@@ -1,19 +1,52 @@
...
+        ...prevState.todos,
+        {
+          userId: prevState.todos.length + 1,
+          id: prevState.todos.length + 1,
+          title,
+          completed: false,
+          user: {
+            name,
+          },
+        },
```
Comment (aliceblunt8) [inline src/App.js:30]: let first create a variable with new movie and then add it to setState call (now code looks messy)

File: src/App.js [hunk 2]
```diff
@@ -1,19 +1,52 @@
...
+  state = {
+    todos: preparedTodos,
+  }
+
+  addNewTodo = (title, name) => {
+    this.setState(prevState => ({
+      todos: [
+        ...prevState.todos,
+        {
+          userId: prevState.todos.length + 1,
```
Comment (aliceblunt8) [inline src/App.js:23]: you need to find userId in users by the user name (it's not a random number)

File: src/App.js [hunk 3]
```diff
@@ -1,19 +1,52 @@
...
 import './App.css';
-
 import users from './api/users';
+import todos from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoForm } from './components/TodoForm/TodoForm';
+
+const preparedTodos = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId),
```
Comment (aliceblunt8) [inline src/App.js:10]: store in todos just userId, no need to save the whole object

File: src/components/TodoForm/TodoForm.jsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    this.setState({
+      title: e.target.value,
+    });
+  };
+
+  selectNameField = (e) => {
+    this.setState({
+      name: e.target.value,
+    });
+  };
```
Comment (aliceblunt8) [inline src/components/TodoForm/TodoForm.jsx:24]: combine this into ONE function

File: src/components/TodoForm/TodoForm.jsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+  addNewTodo = (e) => {
+    e.preventDefault();
+    const { title, name, noTitle, noName } = this.state;
+
+    if (!title || !name) {
+      this.setState({
+        noTitle: !title,
+        noName: !name,
+      });
```
Comment (aliceblunt8) [inline src/components/TodoForm/TodoForm.jsx:34]: if you want convert a primitive value (string) into a boolean you need !! operator

File: src/components/TodoForm/TodoForm.jsx (hunk 4 above)
Comment (aliceblunt8) [reply src/components/TodoForm/TodoForm.jsx:24]: handleChange

File: src/components/TodoForm/TodoForm.jsx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    return (
+      <form onSubmit={this.addNewTodo} className="form">
+
+        <input
+          className="field"
+          type="text"
+          placeholder="type a new todo"
+          value={this.state.title}
+          onChange={this.inputTitleField}
+        />
```
Comment (aliceblunt8) [inline src/components/TodoForm/TodoForm.jsx:60]: add name to the input

File: src/components/TodoForm/TodoForm.jsx [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+              {user.name}
+            </option>
+          ))}
+        </select>
+
+        {this.state.noName && this.state.name === ''
+          ? <span className="error">no user, please select a new one</span>
+          : ''}
+
+        {/* eslint-disable-next-line max-len */}
```
Comment (aliceblunt8) [inline src/components/TodoForm/TodoForm.jsx:87]: why do you disable linter here?

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 8]
```diff
@@ -1,19 +1,56 @@
...
+
+class App extends React.Component {
+  state = {
+    todos: preparedTodos,
+  }
+
+  addNewTodo = (title, name) => {
+    this.setState((prevState) => {
+      const newTodo = {
+        userId: name.id,
```
Comment (aliceblunt8) [inline src/App.js:21]: name is a string, how could it have a prop id?

Comment (aliceblunt8) [reply src/App.js:21]: userId is empty

image

Comment (aliceblunt8) [reply src/App.js:21]: mb save userId to the Todo form state or the whole user and then pass it's userId to the addTodo

File: src/components/TodoForm/TodoForm.jsx (hunk 4 above)
Comment (aliceblunt8) [reply src/components/TodoForm/TodoForm.jsx:24]: fix this

File: src/components/TodoForm/TodoForm.jsx (hunk 6 above)
Comment (aliceblunt8) [reply src/components/TodoForm/TodoForm.jsx:60]: fix this

File: src/components/TodoForm/TodoForm.jsx (hunk 7 above)
Comment (aliceblunt8) [reply src/components/TodoForm/TodoForm.jsx:87]: fix this

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

File: src/components/TodoForm/TodoForm.jsx [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+  //   this.setState({
+  //     title: e.target.value,
+  //   });
+  // };
+  //
+  // selectNameField = (e) => {
+  //   this.setState({
+  //     name: e.target.value,
+  //   });
+  // };
```
Comment (aliceblunt8) [inline src/components/TodoForm/TodoForm.jsx:32]: delete comments

**Review by aliceblunt8 — APPROVED** [review]

**Pull request closed without merging**